src/txn/txn.c

```diff
@@ -1268,6 +1297,8 @@ __wt_txn_init(WT_SESSION_IMPL *session, WT_SESSION_IMPL *session_ret)
 	txn->mod = NULL;
 
 	txn->isolation = session_ret->isolation;
+
+	txn->session_id = session_ret->id;
```

Comment:
I am not resetting the session id when the session is closed. I think since the transaction is part of the session object, it is ok not to reset. Any different opinions?